pkg/termination/termination_test.go

```
@@ -123,66 +123,131 @@ var _ = Describe("Handler Suite", func() {
-				} else {
-					atomic.AddInt32(&counter, 1)
-					rw.Write([]byte(`{"DocumentIncarnation":0,"Events":[]}`))
-				}
+		Context("with Preempt event type", func() {
+			BeforeEach(func() {
+				counter = 0
+
+				// Ensure the polling logic is excercised in tests
+				httpHandler = newMockHTTPHandler(func(rw http.ResponseWriter, req *http.Request) {
+					if atomic.LoadInt32(&counter) == 4 {
+						rw.Write([]byte(`{"DocumentIncarnation":0,"Events":[{"EventType":"Preempt", "ResourceType": "VirtualMachine"}]}`))
```

golangci-lint (2.12.2)

[error] Error return value of `rw.Write` is not checked

(errcheck)